include/sparrow/array_data.hpp

```diff
@@ -412,7 +417,9 @@ namespace sparrow
     template <class U>
     void reference_proxy<L>::update_value(U&& u)
     {
-        p_layout->update(m_index, std::forward<U>(u));
+        m_bit_ref = true;
```

Comment:
If the value is null, then `m_bit_ref = false` isnt it?

Reply:
This method updates the inner value, not the value of the array (for this one, there is `update`)